bots/pr/src/main/java/org/openjdk/skara/bots/pr/JEPCommand.java

```diff
@@ -0,0 +1,163 @@
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+public class JEPCommand implements CommandHandler {
+    static final String JEP_LABEL = "jep";
+    private static final Pattern issuePattern = Pattern.compile("(([A-Za-z]+-)?[0-9]+)(?: |$)");
+    private static final String jepMarker = "<!-- jep: '%s' '%s' '%s' -->"; // <!-- jep: 'JEP-ID' 'ISSUE-ID' 'ISSUE-TITLE' -->
+    static final Pattern jepMarkerPattern = Pattern.compile("<!-- jep: '(.*?)' '(.*?)' '(.*?)' -->");
+    private static final String unneededMarker = "<!-- jep: 'unneeded' 'unneeded' 'unneeded' -->";
+
+    private void showHelp(PrintWriter reply) {
+        reply.println("""
+                Command syntax:
+                 * `/jep <issue-id>`
+                 * `/jep JEP-<jep-id>`
+                 * `/jep jep-<jep-id>`
+                 * `/jep [unneeded|uneeded]`
+
+                Some examples:
+
+                 * `/jep JDK-1234567`
+                 * `/jep 1234567`
+                 * `/jep jep-123`
+                 * `/jep JEP-123`
+                 * `/jep uneeded`
+                 * `/jep unneeded`
+
+                Note:
+                The project prefix of the issue ID (`JDK-` in the above examples) is optional.
+                But the `JEP-` or `jep-` prefix is required when you provide a JEP ID.
+                And the issue type must be `JEP`.
+                """);
+    }
+
+    private Optional<Issue> getJepIssue(String args, PullRequestBot bot, PrintWriter reply) {
+        Optional<Issue> jbsIssue;
+        if (args.startsWith("jep-") || args.startsWith("JEP-")) {
+            // Handle the JEP ID
+            jbsIssue = bot.issueProject().jepIssue(args.substring(4));
+        } else {
+            // Handle the issue ID
+            jbsIssue = bot.issueProject().issue(args);
+        }
+        return jbsIssue;
+    }
+
+    @Override
+    public void handle(PullRequestBot bot, PullRequest pr, CensusInstance censusInstance, Path scratchPath,
+                       CommandInvocation command, List<Comment> allComments, PrintWriter reply) {
+        if (!pr.author().equals(command.user()) && !censusInstance.isReviewer(command.user())) {
+            reply.println("only the pull request author and [Reviewers](https://openjdk.java.net/bylaws#reviewer) are allowed to use the `jep` command.");
+            return;
+        }
+
+        var args = command.args().trim();
+        if (args.isEmpty() || args.isBlank()) {
+            showHelp(reply);
+            return;
+        }
+
+        var labelNames = pr.labelNames();
+        if ("unneeded".equals(args) || "uneeded".equals(args)) {
+            if (pr.author().equals(command.user()) && !censusInstance.isReviewer(command.user())) {
+                reply.println("only [Reviewers](https://openjdk.java.net/bylaws#reviewer) can determine that a JEP request is not needed.");
+                return;
+            }
```

Comment:
I don't think I agree with this check. If the author is able to link the PR to a JEP, then the author should be able to correct a mistakenly linked JEP.

Reply:
The author can use the `/jep <jep-id>|<issue-id>` again if the author linked the wrong JEP. This check follows the convention of the `csr` command. It can protect the situation that the author uses the command `/jep unneeded` and integrates the patch unexpectly.

Actually, the `/jep unneeded` command is rarely used and the author can request the reviewers to do that if the patch really don't need a JEP to be targeted. Preventing the unexpected (and maybe wrong) code from integrating should be the most important thing we need to ensure even though it has not happened in the past.

Reply:
I understand that this is how it would work with the current logic, and I don't agree with it. Declaring the need for a JEP and a CSR are different, so they do not need to follow the same model (though I'm not sure I agree with the limitation on `/csr unneeded` either).

In order for a committer to mishandle this, they would have to actively issue `/jep unneeded` and then integrate. I think it's unlikely to happen by mistake, and mistakes is what this feature intends to protect against. We already trust committers to handle `/integrate` on their own, this isn't about protecting against malicious committers.

Reply:
> We already trust committers to handle /integrate on their own, this isn't about protecting against malicious committers.

I agree with you now and I will fix it later.

Reply:
> I understand that this is how it would work with the current logic, and I don't agree with it. Declaring the need for a JEP and a CSR are different, so they do not need to follow the same model (though I'm not sure I agree with the limitation on `/csr unneeded` either).

I feel that the current limitation on `/csr unneeded` is a good one, but I also agree that JEP need not (and probably should not) have this same restriction.

Reply:
Fixed just now.